### PR TITLE
Bump minimum required Python to 3.7

### PR DESCRIPTION
**Pull request by mabl**
Since Python 2.7 is end of life, bump support to Python 3
only. The oldest version with active full support is python
3.7.  This also gives us support for glorious f-strings.

### CODE REVIEW

File: .binstar.yml [hunk 1]
```diff
@@ -5,9 +5,8 @@ platform:
   - linux-32
 
 engine:
-  - python=2.7
-  - python=3.4
-  - python=3.5
+  - python=3.7
+  - python=3.7
```
Comment (jordens) [inline .binstar.yml:9]:
```suggestion
  - python=3.8
```

File: rayopt/simplex.py [hunk 2]
```diff
@@ -192,5 +189,5 @@ def transform(self, t):
             p = simplex_transform(self.i.ravel(), self.j, self, t)
             return p.view(self.__class__)
 
-    Simplex.__name__ = str("Simplex{}d{}n".format(d0, n0))
+    Simplex.__name__ = str(f"Simplex{d0}d{n0}n")
```
Comment (jordens) [inline rayopt/simplex.py:192]: I don't remember why that `str()` was there. It seems weird.

**Review by jordens — CHANGES_REQUESTED** [review]
Wohoo! Thanks a lot.

**Comment by jordens** [comment]
Thanks for the great work. If you feel like contributing more, please add yourself to authors and I'll add you to the project.

File: rayopt/simplex.py (hunk 2 above)
Comment (mabl) [reply rayopt/simplex.py]: Removed.

File: .binstar.yml (hunk 1 above)
Comment (mabl) [reply .binstar.yml]: Fixed

**Comment by mabl** [comment]
> Thanks for the great work. If you feel like contributing more, please add yourself to authors and I'll add you to the project.

For now, my contributions are minor. No need to add me to the authors. 

I'm looking into geometrical optics these days, and evaluate the state of python based ray tracers. This is the one which worked first try for me, and your extensive list of examples is great! I could directly start and use it to trace a proprietary Zemax file I had to analyse. And I even directly found a flaw in the optics. Great!

I do think that the code could benefit from a some cleanup, so my plan is to dig through it and understand as much as possible :-) Let's see how it goes.

**Pull request merged**